File: android/src/main/java/space/luas/rnsharedprefs/RNSharedPrefsPackage.java
package space.luas.rnsharedprefs;

import androidx.annotation.NonNull;

import com.facebook.react.BaseReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.model.ReactModuleInfo;
import com.facebook.react.module.model.ReactModuleInfoProvider;

import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * React Native package for Foreground Service TurboModule
 *
 * This package registers the ForegroundServiceModule with React Native.
 * Extends TurboReactPackage for TurboModule support.
 */
public class RNSharedPrefsPackage extends BaseReactPackage {

    @Nullable
    @Override
    public NativeModule getModule(String name, @Nonnull ReactApplicationContext reactContext) {
        if (name.equals(RNSharedPrefsModule.NAME)) {
            return new RNSharedPrefsModule(reactContext);
        }
        return null;
    }

    @NonNull
    @Override
    public ReactModuleInfoProvider getReactModuleInfoProvider() {
        return new ReactModuleInfoProvider() {
            @NonNull
            @Override
            public Map<String, ReactModuleInfo> getReactModuleInfos() {
                Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
                moduleInfos.put(RNSharedPrefsModule.NAME,
                    new ReactModuleInfo(
                        RNSharedPrefsModule.NAME,       // name
                        RNSharedPrefsModule.class.getName(),  // className
                        false, // canOverrideExistingModule
                        true, // needsEagerInit
                        false, // isCXXModule
                        true   // isTurboModule
                    )
                );
                return moduleInfos;
            }
        };
    }
}

File: android/src/main/java/com/luas/rnsharedprefs/RnSharedPrefsPackage.kt
package com.luas.rnsharedprefs

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider
import java.util.HashMap

class RnSharedPrefsPackage : BaseReactPackage() {
  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == RnSharedPrefsModule.NAME) {
      RnSharedPrefsModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider() = ReactModuleInfoProvider {
    mapOf(
      RnSharedPrefsModule.NAME to ReactModuleInfo(
        name = RnSharedPrefsModule.NAME,
        className = RnSharedPrefsModule.NAME,
        canOverrideExistingModule = false,
        needsEagerInit = false,
        isCxxModule = false,
        isTurboModule = true
      )
    )
  }
}
